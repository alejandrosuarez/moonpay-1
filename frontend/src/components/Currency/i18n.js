const translations = {
  fr: {
    tagUS: "US",
    tagTestMode: "Mode Test",
  },
  en: {
    tagUS: "US",
    tagTestMode: "Test Mode",
  },
};

export default translations;
