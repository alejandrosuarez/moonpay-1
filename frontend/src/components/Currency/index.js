import React from "react";
import { SORT_BY_NAME } from "../../settings/constants";
import { Box, Text } from "grommet";
import "./index.scss";
import translations from "./i18n";
import { useSelector } from "react-redux";

const Currency = ({ currency = {}, sortBy }) => {
  // Get local from redux state for the translations
  const { locale } = useSelector((state) => state.locale);

  return (
    <>
      {sortBy === SORT_BY_NAME ? (
        <Text>
          {currency?.code} - <b>{currency?.name?.toUpperCase()}</b>
        </Text>
      ) : (
        <Text data-testid="title-code">
          <b>{currency?.code?.toUpperCase()}</b> - {currency?.name}
        </Text>
      )}
      <Box direction="row" align="center" justify="center" gap="small">
        {currency.isSupportedInUS && (
          <Box
            size="small"
            pad={{ horizontal: "small" }}
            background="accent-3"
            round="small"
          >
            <Text size="xsmall" data-testid="tag-us">
              {translations[locale].tagUS}
            </Text>
          </Box>
        )}
        {currency.supportsTestMode && (
          <Box
            size="small"
            background="accent-1"
            round="small"
            pad={{ horizontal: "small" }}
          >
            <Text size="xsmall" data-testid="tag-test-mode">
              {translations[locale].tagTestMode}
            </Text>
          </Box>
        )}
      </Box>
    </>
  );
};

export default Currency;
