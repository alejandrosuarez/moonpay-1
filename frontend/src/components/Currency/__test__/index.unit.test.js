import React from "react";
import ReactDom from "react-dom";
import Currency from "../index";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import { create } from "react-test-renderer";

jest.mock("react-redux", () => ({
  useSelector: () => ({ locale: "en" }),
}));

describe("Currency Component", () => {
  afterEach(cleanup);

  it("Renders without crashing", () => {
    const div = document.createElement("div");
    ReactDom.render(<Currency />, div);
  });

  it("Renders currency correctly", () => {
    const currency = {
      code: "test",
      name: "Test",
      isSupportedInUS: true,
      supportsTestMode: true,
    };
    const sortBy = "code";
    const { getByTestId } = render(
      <Currency currency={currency} sortBy={sortBy} />
    );
    expect(getByTestId("tag-us")).toHaveTextContent("US");
    expect(getByTestId("tag-test-mode")).toHaveTextContent("Test Mode");
    expect(getByTestId("title-code")).toHaveTextContent(
      `${currency.code.toUpperCase()} - ${currency.name}`
    );
  });

  it("Matches snapshot empty", () => {
    const currency = create(<Currency />).toJSON();
    expect(currency).toMatchSnapshot();
  });

  it("Matches snapshot with props", () => {
    const currency = {
      code: "test",
      name: "Test",
      isSupportedInUS: true,
      supportsTestMode: true,
    };
    const sortBy = "code";
    const currencyElement = create(
      <Currency currency={currency} sortBy={sortBy} />
    ).toJSON();
    expect(currencyElement).toMatchSnapshot();
  });
});
