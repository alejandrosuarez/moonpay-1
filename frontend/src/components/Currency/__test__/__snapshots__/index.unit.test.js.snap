// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Currency Component Matches snapshot empty 1`] = `
Array [
  <span
    className="StyledText-sc-1sadyjn-0 DkZvS"
    data-testid="title-code"
  >
    <b />
     - 
  </span>,
  <div
    className="StyledBox-sc-13pk1d4-0 iQEJEP"
  />,
]
`;

exports[`Currency Component Matches snapshot with props 1`] = `
Array [
  <span
    className="StyledText-sc-1sadyjn-0 DkZvS"
    data-testid="title-code"
  >
    <b>
      TEST
    </b>
     - 
    Test
  </span>,
  <div
    className="StyledBox-sc-13pk1d4-0 iQEJEP"
  >
    <div
      className="StyledBox-sc-13pk1d4-0 lklmjx"
      size="small"
    >
      <span
        className="StyledText-sc-1sadyjn-0 gJdaHr"
        data-testid="tag-us"
      >
        US
      </span>
    </div>
    <div
      className="StyledBox__StyledBoxGap-sc-13pk1d4-1 eBVJAG"
    />
    <div
      className="StyledBox-sc-13pk1d4-0 gZyYMm"
      size="small"
    >
      <span
        className="StyledText-sc-1sadyjn-0 gJdaHr"
        data-testid="tag-test-mode"
      >
        Test Mode
      </span>
    </div>
  </div>,
]
`;
